### DISCUSSION TITLE
Data types

### DISCUSSION THREAD

**Discussion by PeterW-DASA** [m-ld/m-ld-spec#44] [Discussion]
What data types does the approach work with? Text only, or images/video also?

**Comment by gsvarovsky** [comment]
The optimal data type for **m-ld** is structured records. A rule of thumb would be, if you can sensibly encode and manipulate your data as JSON, it suits **m-ld** well. This covers an enormous area of possible data structures. However it would not be a good choice for images or video; these can be embedded in JSON fields, but not very efficiently, and smart synchronisation of these fields would be separate to **m-ld**.

Same actually goes for large blocks of text. We're looking at smart synchronisation of text soon. There's a number of possible clever algorithms developed in this field which we could use, and the choice is going to be something that you declare in the data itself, to best suit the use-case. Also a topic of an upcoming blog post!
